pkg/bundle/fs.go

```diff
@@ -0,0 +1,199 @@
+package bundle
+
+import (
+	"io/fs"
+	"path/filepath"
+	"strings"
+
+	operatorsv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	rukpakv1alpha1 "github.com/operator-framework/rukpak/api/v1alpha1"
+)
+
+// FS is a File System that contains a bundle.
+//
+// A bundle is a directory that contains files that may or may not be
+// Kubernetes manifests. When opening a manifest file, it will return an
+// `ObjectFile` type that allows the consumer to access the typed objects
+// without first opening/parsing the manifest itself. It still fully supports
+// all file operations.
+//
+// TODO(ryantking): Should we support any other file system abstractions beyond `fs.FS`?
+// `fs.FS` doesn't support any write operations. They can modify the objects in an `ObjectFile`, but that won't change
```

Comment:
IMO, fs.FS is sufficient. We can use https://pkg.go.dev/testing/fstest#MapFS as a writable in-memory FS implementation, and if we want a writable disk implementation, then we can just use `os.*` functions to create the necessary files/directories and then use `os.DirFS` to read it.